frame/header-mmr/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+//! # Chain MMR Pallet
+//!
+//! \## Overview
+//! This is the pallet to maintain accumulate headers Merkle Mountain Range
+//! and push the mmr root in to the digest of block headers on finalize.
+//! MMR can be used for light client to implement super light clients,
+//! and can also be used in other chains to implement chain relay for
+//! cross-chain verification purpose.
+//!
+//! \## Terminology
+//!
+//! \### Merkle Moutain Range
+//! For more details about the MMR struct, refer https://github.com/mimblewimble/grin/blob/master/doc/mmr.md#structure
+//!
+//! \### MMR Proof
+//! Using the MMR Store Storage, MMR Proof can be generated for specific
+//! block header hash. Proofs can be used to verify block inclusion together with
+//! the mmr root in the header digest.
+//!
+//! \### Positions
+//! The index position of the nodes(and hash leave nodes) in the mmr node list
+//! constructed using MMR struct
+//!
+//! \### Digest Item
+//! The is a ```MerkleMountainRangeRoot(Hash)``` digest item pre-subscribed in Digest.
+//! This is implemented in Darwinia's fork of substrate: https://github.com/darwinia-network/substrate
+//! The Pull request link is https://github.com/darwinia-network/substrate/pull/1
+//!
+//! \## Implementation
+//! We are using the MMR library from https://github.com/nervosnetwork/merkle-mountain-range
+//! Pull request: https://github.com/darwinia-network/darwinia/pull/358
+//!
+//! \## References
+//! Darwinia Relay's Technical Paper:
+//! https://github.com/darwinia-network/rfcs/blob/master/paper/Darwinia_Relay_Sublinear_Optimistic_Relay_for_Interoperable_Blockchains_v0.7.pdf
+//!
+//! https://github.com/mimblewimble/grin/blob/master/doc/mmr.md#structure
+//! https://github.com/mimblewimble/grin/blob/0ff6763ee64e5a14e70ddd4642b99789a1648a32/core/src/core/pmmr.rs#L606
+//! https://github.com/nervosnetwork/merkle-mountain-range/blob/master/src/tests/test_accumulate_headers.rs
+//! https://eprint.iacr.org/2019/226.pdf
+//!
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{decl_error, decl_event, decl_module, decl_storage, ensure};
+use sp_std::marker::PhantomData;
+use sp_std::prelude::*;
+//use frame_benchmarking::{benchmarks, account};
+use frame_system::{self as system}; // , ensure_root, ensure_signed, RawOrigin
+
+//use codec::{Encode, Decode};
+use sp_runtime::{
+	generic::DigestItem,
+	traits::{Hash, One},
+	DispatchError,
+};
+
+use merkle_mountain_range::{MMRStore, MerkleProof, MMR};
+
+mod mock;
```

Comment:
cfg missing

Reply:
```#![cfg(test)]```

It is defined in mock.rs and test.rs, follows the convention of the sample pallet in substrate

Reply:
https://github.com/paritytech/substrate/blob/v2.0.0-alpha.4/frame/elections/src/mock.rs#L19

https://github.com/paritytech/substrate/blob/v2.0.0-alpha.4/frame/elections/src/lib.rs#L42